src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_VALUE = 10;
+    private static final double MORE_SIZE = 1.5;
+
+    private T[] array;
+
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_VALUE];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        array[size] = value;
+        size++;
```

Comment:
```suggestion
        array[size++] = value;
```
not a mistake but FYI